x/stakeibc/keeper/reward_converter.go

```diff
@@ -57,75 +57,95 @@ type FeeInfo struct {
+		return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed,
 			"community pool liquid staked amount greater than total delegations")
 	}
 
 	// The rebate amount is determined by the contribution of the community pool stake towards the total TVL,
 	// multiplied by the rebate fee percentage
 	contributePercentage := sdk.NewDecFromInt(rebateInfo.LiquidStakeAmount).Quo(sdk.NewDecFromInt(hostZone.TotalDelegations))
-	rebateAmount := sdk.NewDecFromInt(totalFeesAmount).Mul(contributePercentage).Mul(rebateInfo.RebatePercentage).TruncateInt()
-	strideFeeAmount := totalFeesAmount.Sub(rebateAmount)
+	totalFeesAmount := sdk.NewDecFromInt(rewardAmount).Mul(strideFeePercentage).TruncateInt()
+	rebateAmount = sdk.NewDecFromInt(totalFeesAmount).Mul(contributePercentage).Mul(rebateInfo.RebatePercentage).TruncateInt()
+	remainingAmount = rewardAmount.Sub(rebateAmount)
 
-	// Sanity check calculation
-	if !rebateAmount.Add(strideFeeAmount).Add(reinvestAmount).Equal(rewardAmount) {
-		return feeInfo, errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed, "fee info compontents do not sum to rewards")
-	}
+	return rebateAmount, remainingAmount, nil
+}
 
-	feeInfo = FeeInfo{
-		RebateAmount:    rebateAmount,
-		StrideFeeAmount: strideFeeAmount,
-		ReinvestAmount:  reinvestAmount,
+// Given the reward balance from an ICQ, calculates the portions intended as a stride fee vs for reinvestment
+//
+// If the chain doesn't have a rebate in place, the split is decided entirely from the stride commission percent
+// However, if the chain does have a rebate, we need to factor that into the calculation
+//
+// For instance, if 100 rewards were collected and 5 were sent as a rebate, then the stride fee should be based
+// on the original 100 rewards instead of the remaining 95
+func (k Keeper) CalculateRewardsFeeSplit(
+	ctx sdk.Context,
+	hostZone types.HostZone,
+	rewardsAmount sdkmath.Int,
+) (strideFeeAmount sdkmath.Int, reinvestAmount sdkmath.Int) {
+	// Get the fee percentage from params
+	strideFee := sdk.NewIntFromUint64(k.GetParams(ctx).StrideCommission)
+	totalFeePercentage := sdk.NewDecFromInt(strideFee).Quo(sdk.NewDec(100))
+	totalFeesAmount := sdk.NewDecFromInt(rewardsAmount).Mul(totalFeePercentage)
+
+	// Check if the chain has a rebate
+	rebateInfo, chainHasRebate := hostZone.SafelyGetCommunityPoolRebate()
+
+	// If there's no rebate, the fee split just uses the commission
+	if !chainHasRebate {
+		strideFeeAmount = totalFeePercentage.Mul(sdk.NewDecFromInt(rewardsAmount)).TruncateInt()
+	} else {
+		// Otherwise, the rebate must be considered in the fee split
+		// The rebate portion the portion of TVL contributed to by the liquid stake * the rebate percentage
+		// The stride fee poriton is the remaining percentage
+		contributePercentage := sdk.NewDecFromInt(rebateInfo.LiquidStakeAmount).Quo(sdk.NewDecFromInt(hostZone.TotalDelegations))
+		rebatePercentage := contributePercentage.Mul(rebateInfo.RebatePercentage)
```

Comment:
nit: confusing that this var and rebateInfo.RebatePercentage have the same name but represent different concepts

suggest renaming this

maybe `RebatePctOfAllZoneRewards` or similar

Reply:
lgtm